docs/contributor/arch/029-batch-processor-migration-to-exporterhelper.md

```diff
@@ -0,0 +1,219 @@
+delivery to all backends, causing:
+
+- Significantly slower telemetry delivery to healthy backends
+- Potential data loss when client queues fill up
+- Duplicated telemetry showing up in the healthy backend due to client retries
+- Overall system degradation despite having healthy backends available
+
+## Decision
+
+We will migrate from `batchprocessor` to the exporterhelper's built-in batching capabilities for all exporter
+configurations. During this migration period, we will try to get in touch with the OpenTelemetry community to discuss
+and potentially contribute a solution.
+The [design document](https://docs.google.com/document/d/1uxnn5rMHhCBLP1s8K0Pg_1mAs4gCeny8OWaYvWcuibs) proposed a
+`drop_on_error` configuration in the exporter helper package to enable users to select which pipelines should not
+propagate backpressure. However, there are currently no updates on the progress of this proposal. We will try to
+personally contact one of the OpenTelemetry maintainers to discuss this further.
+
+Currently, the migration plan includes the following steps to mitigate the fanout backpressure issue:
+
+### Backpressure Isolation: Custom Processor/Connector for Primary/Secondary Backend Designation
+
+We will implement a custom processor or connector that allows designation of primary and secondary backends:
+
+- **Primary backends**: Backpressure propagates normally when these backends are unhealthy
+- **Secondary backends**: Operate asynchronously without propagating backpressure to the main pipeline
+- **API exposure**: Users can configure primary backends through the Telemetry CR
+
+An example configuration for the custom processor may look like this:
+
+```yaml
+processors:
+  fanout:
+    primary_exporters: [ otlp/backend ]
+    secondary_exporters: [ otlp/backend-2, file/backend ]
+
+exporters:
+  otlp/backend:
+    endpoint: backend:4317
+    sending_queue:
+      enabled: true
+      queue_size: 500
+    retry_on_failure:
+      enabled: true
+
+  otlp/backend-2:
+    endpoint: unhealthy-backend:4317
+    sending_queue:
+      enabled: true
+      queue_size: 500
+    retry_on_failure:
+      enabled: true
+
+service:
+  pipelines:
+    traces:
+      receivers: [ otlp ]
+      processors: [ fanout ]
+      exporters: [ otlp/backend, otlp/backend-2, file/backend ]
+```
+
+### Deduplication : Custom Processor for UUIDv5-based Deduplication
+
+To handle potential duplicate telemetry in secondary backends resulting from client retries, we will implement a custom
+processor that generates UUIDv5 identifiers for each telemetry item based on its:
+
+- Timestamp
+- Pod Name
+- Node Name
+- Log Body (for logs)
+- Span ID (for traces)
+- Metric Name (for metrics)
+
+This processor will ensure idempotent delivery by allowing backends to recognize and discard duplicates.
+
+### Monitoring Implementation
+
+The following metrics will be monitored via self-monitoring:
```

Comment:
Are you referring to the [self monitor component](https://help.sap.com/docs/btp/sap-business-technology-platform/telemetry-architecture?locale=en-US#self-monitor)? Or is there another method of self-monitoring?
(I'm trying to convert this to active voice, so it's helpful to know who/what does the self-monitoring).

```suggestion
Our self monitor component monitors the following metrics:
```